### PR TITLE
Move development dependencies to `pyproject.toml`

### PR DESCRIPTION
**Pull request by sbrugman**
Minor PR to kick-off contributing to `torchfix`. I'm developing new functionality, which I plan to contribute in follow-up PRs. 

Changes:
- Move `requirements-dev.txt` to `optional-dependencies`
- Explicitly set the setuptools build system
- Omit `pip install pre-commit` in CONTRIBUTING.md, already included in dev dependencies

### CODE REVIEW

**Comment by kit1980** [comment]
Thanks!
Something is not right, as pytest is now not found.

**Comment by sbrugman** [comment]
Thanks for the quick reply! Was able to resolve the issues with the CI/CD feedback:

- Ensured the build dependencies are up-to-date
- Fixed the automatic package discovery config to pick up subpackages (`torchfix.visitors`)
- Added the `__init__.py` file to `torch.visitors` to make it a proper [namespace package](https://docs.python.org/3/reference/import.html#regular-packages)

File: pyproject.toml [hunk 1]
```diff
@@ -50,8 +50,8 @@ pythonpath = [
 [tool.black]
 exclude = "tests/fixtures/*"
 
-[tool.setuptools]
-packages = ["torchfix"]
+[tool.setuptools.packages.find]
+include = ["torchfix*"]
```
Comment (kit1980) [inline pyproject.toml:54]: Could you explain why is this needed?

Comment (sbrugman) [reply pyproject.toml]: It's not needed, I removed it. 

(In my local code base another directory with feature development was causing `setuptools` to discover multiple top-level packages, but that's unrelated to this PR)

**Review by kit1980 — APPROVED** [review]

**Comment by sbrugman** [comment]
Thanks!
